### ISSUE TITLE
False positive for Lint/FormatParameterMismatch when options param is a local variable

### ISSUE THREAD

**Issue by backus** [bbatsov/rubocop#2506]
Example

```ruby
params = { y: '2015', m: '01', d: '01' }

puts format('%{y}-%{m}-%{d}', params)
```

Rubocop output

```sh
$ cat ex.rb
params = { y: '2015', m: '01', d: '01' }

puts format('%{y}-%{m}-%{d}', params)

$ ruby ex.rb
2015-01-01

$ rubocop -v
0.35.1

$ rubocop -D ex.rb
Inspecting 1 file
W

Offenses:

ex.rb:3:6: W: Lint/FormatParameterMismatch: Number arguments (1) to format mismatches expected fields (0).
puts format('%{y}-%{m}-%{d}', params)
     ^^^^^^

1 file inspected, 1 offense detected
```

**Comment by DelmerGA** [comment]
:+1: just had this issue

**Comment by alexdowad** [comment]
The thing to check here is whether the format string contains `%{name}` or `%<name>` format specifiers.

**Comment by alexdowad** [comment]
Just pushed a fix to a branch which has an open PR...

**Comment by backus** [comment]
Wow thanks for the quick fix @alexdowad

**Comment by backus** [comment]
@alexdowad I cloned your fork and verified that the issue in question is fixed on your `perf` branch. Thanks again. I think it might make sense though to cherry-pick this fix out of your perf branch and open a separate PR for it since this is independent of performance linting and also likely to be merged and released much more quickly than a new set of features.
